src/app/core/data/entries-response-parsing.service.ts

```diff
@@ -0,0 +1,42 @@
+import { isNotEmpty } from '../../shared/empty.util';
+import { ObjectCacheService } from '../cache/object-cache.service';
+import { ErrorResponse, GenericSuccessResponse, RestResponse } from '../cache/response.models';
+import { DSpaceRESTV2Response } from '../dspace-rest-v2/dspace-rest-v2-response.model';
+import { DSpaceSerializer } from '../dspace-rest-v2/dspace.serializer';
+import { BaseResponseParsingService } from './base-response-parsing.service';
+import { ResponseParsingService } from './parsing.service';
+import { RestRequest } from './request.models';
+import { CacheableObject } from '../cache/object-cache.reducer';
+import { GenericConstructor } from '../shared/generic-constructor';
+
+export abstract class EntriesResponseParsingService<T extends CacheableObject> extends BaseResponseParsingService implements ResponseParsingService {
```

Comment:
I'm confused by the purpose of this ParsingService.  It seems like you are using this for two entirely different services which just happen to both have an `/entries` endpoint...that seems like an odd design choice to me.  Essentially, if I'm understanding correct, this same parser is used for

* Browse By entries like: `/discover/browses/author/entries`
* Vocabulary entries like: `/submission/vocabularies/<:vocabulary-name>/entries`

It seems really odd to me to treat these endpoints with the same parser, as they are unrelated (as far as I can tell), and just by coincidence both have a `/entries` subpath.

Am I missing something here?  Is there as better way to implement this parser so that it's more based on the object type and less based on a coincidental path name?  Maybe @artlowel has ideas of how to improve this?

Reply:
Since the entries don't have a self link, the [buildList](https://github.com/4Science/dspace-angular/blob/c8c7e927173f6139cc26174e65c059519c4ee908/src/app/core/cache/builders/remote-data-build.service.ts#L130) method can't build paginated list properly. To work around to this issue browse service has a specific response parser. Since the solution for vocabulary service is the same to not duplicate the code I've created an abstract `EntriesResponseParsingService` which is extended by [BrowseEntriesResponseParsingService](https://github.com/4Science/dspace-angular/blob/c8c7e927173f6139cc26174e65c059519c4ee908/src/app/core/data/browse-entries-response-parsing.service.ts#L8) and [VocabularyEntriesResponseParsingService](https://github.com/4Science/dspace-angular/blob/c8c7e927173f6139cc26174e65c059519c4ee908/src/app/core/submission/vocabularies/vocabulary-entries-response-parsing.service.ts#L12)

Reply:
After the meeting, I looked back at the `/vocabularies/<:vocabular-name>/entries` endpoint.  @atarix83's point is correct, currently the `entries` do not have a `self` link: https://github.com/DSpace/Rest7Contract/blob/main/vocabularies.md#controlled-vocabulary-entries

This `self` link doesn't seem possible to add as there is not a unique identifier for all types of entries. Some (like the SRSC vocabulary) have an `id`, but others (like `common_types` from a submission config) do not.  So, I now see why @atarix83 is noting that he had to take the above approach for this `/entries` path, as they are not always full objects, sometimes they are just lists of values.